### ISSUE TITLE
GPU-Util is zero

### ISSUE THREAD

**Issue by purse1996** [jmiller656/EDSR-Tensorflow#30]
Thank you for your sharing code. I met a problem. When I train the model for my data in GPU,the memory-usage is high but the GPU-util is always zero. So training speed is slow. It confuse me very much.

**Comment by jmiller656** [comment]
Not sure what this means, I haven't experienced this error. Could you give me some steps to reproduce the error?

**Comment by xinqingbuhao** [comment]
![image](https://user-images.githubusercontent.com/37792713/57631347-e9a48a00-75d1-11e9-8c54-96b0b5d3a6a5.png)
i got the same issue plus resource-exhausted.

**Comment by xinqingbuhao** [comment]
> Not sure what this means, I haven't experienced this error. Could you give me some steps to reproduce the error?

 11%|███▍                             | 1058/10000 [9:15:04<77:40:33, 31.27s/it]

Is this state normal with GPU ?  I think this speed is a bit slow and GPU-Util is zero with 10000 img of 100*100, batchsize=32 ,32layer Net. thx~~

**Comment by jmiller656** [comment]
Hey there, I believe you are right that the timing you show is quite slow. I think there may be a problem with your cuda setup, unrelated to the code here. This would probably cause zero GPU utilization. Nothing in this project directly edits what is being used in the gpu. That is done behind the scenes by tensorflow

**Comment by xinqingbuhao** [comment]
> Hey there, I believe you are right that the timing you show is quite slow. I think there may be a problem with your cuda setup, unrelated to the code here. This would probably cause zero GPU utilization. Nothing in this project directly edits what is being used in the gpu. That is done behind the scenes by tensorflow

thx~~little brother ,hahah, Could u tell me your tensorflow-version , CUDA and cudnn version? （i use python2.7 env）

**Comment by xinqingbuhao** [comment]
i solved the issue. The reason is why data  is not prepared when GPU need to compute. Now i put the data and label into  files, GPU can get data immediately when the training stage.
Although your code decouples data processing and training networks. But it caused serial training instead of parallelism, and the main expenses were placed on the cpu. It is possible when the amount of data is small, but it is extremely inefficient when the data is slightly larger.

**Comment by jmiller656** [comment]
I see, good catch. Do you have code for this? If so, please make a PR and I'll review it
